lib/src/test/java/uk/gov/di/ipv/cri/address/library/service/AddressServiceTest.java

```diff
@@ -143,4 +151,330 @@ void shouldGetAddressItem() {
+
+        // Order Reversed
+        List<CanonicalAddress> canonicalAddresses = List.of(previousAddress, currentAddress);
+
+        addressService.setAddressValidity(canonicalAddresses);
+
+        assertTrue(addressService.isCurrentAddress(currentAddress));
+        assertTrue(addressService.isNotCurrentAddress(previousAddress));
+        assertNotNull(previousAddress.getValidUntil());
+    }
+
+    @Test
+    void testAddressesLinkerFailsIfGivenInvalidFirstAddressDates()
+            throws AddressProcessingException {
+
+        LocalDate date = LocalDate.of(2013, 8, 9);
+
+        CanonicalAddress address0 = new CanonicalAddress();
+        address0.setValidFrom(date);
+        address0.setValidUntil(null);
+
+        CanonicalAddress address1 = new CanonicalAddress();
+        address1.setValidFrom(date);
+        address1.setValidUntil(date);
+
+        assertTrue(addressService.isCurrentAddress(address0));
+        assertTrue(addressService.isNotCurrentAddress(address1));
+
+        List<CanonicalAddress> canonicalAddresses = List.of(address0, address1);
+
+        AddressProcessingException exception =
+                assertThrows(
+                        AddressProcessingException.class,
+                        () -> addressService.setAddressValidity(canonicalAddresses));
+
+        assertEquals(AddressService.ERROR_ADDRESS_DATE_IS_INVALID, exception.getMessage());
+        assertTrue(addressService.isCurrentAddress(address0));
+        assertTrue(addressService.isNotCurrentAddress(address1));
+        assertNotNull(address1.getValidFrom());
+        assertNotNull(address1.getValidUntil());
+    }
+
+    @Test
+    void testAddressesLinkerFailsIfGivenInvalidSecondAddressDates()
+            throws AddressProcessingException {
+
+        LocalDate date = LocalDate.of(2013, 8, 9);
+
+        CanonicalAddress address0 = new CanonicalAddress();
+        address0.setValidFrom(date);
+        address0.setValidUntil(null);
+
+        CanonicalAddress address1 = new CanonicalAddress();
+        address1.setValidFrom(date);
+        address1.setValidUntil(date);
+
+        assertTrue(addressService.isCurrentAddress(address0));
+        assertTrue(addressService.isNotCurrentAddress(address1));
+
+        // Reversed
+        List<CanonicalAddress> canonicalAddresses = List.of(address1, address0);
+
+        AddressProcessingException exception =
+                assertThrows(
+                        AddressProcessingException.class,
+                        () -> addressService.setAddressValidity(canonicalAddresses));
+
+        assertEquals(AddressService.ERROR_ADDRESS_DATE_IS_INVALID, exception.getMessage());
+        assertTrue(addressService.isCurrentAddress(address0));
+        assertTrue(addressService.isNotCurrentAddress(address1));
+        assertNotNull(address1.getValidFrom());
+        assertNotNull(address1.getValidUntil());
+    }
+
+    @Test
+    void testAddressesLinkerThreeAddressesFail() {
```

Comment:
rename to something along the lines of shouldThrowExceptionWhenThreeAddressesProvided

Reply:
All the tests have been renamed to the format methodname, what should happen,  given what condition.